cnf-certification-test/lifecycle/suite.go

```diff
@@ -584,23 +582,25 @@ func TestPodTolerationBypass(env *provider.TestEnvironment) {
 	testhelper.AddTestResultLog("Non-compliant", podsWithRestrictedTolerationsNotDefault, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 func testStorageRequiredPods(env *provider.TestEnvironment) {
-	oc := clientsholder.GetClientsHolder()
-	var nonCompliantPods []*provider.Pod
-	for _, pod := range env.Pods {
-		for _, podOwner := range pod.ObjectMeta.GetOwnerReferences() {
-			if podOwner.Kind != statefulSet {
-				continue
-			}
-			statefulset, err := oc.K8sClient.AppsV1().StatefulSets(pod.Namespace).Get(context.TODO(), podOwner.Name, apiv1.GetOptions{})
-			if err != nil {
-				tnf.ClaimFilePrintf(err.Error())
+	var persistentVolumesBad []string
+	if env.StorageList[index] == "kubernetes.io/no-provisioner" {
+		tnf.ClaimFilePrintf("The class storage is using local storage")
+		return
+	}
+
+	for _, put := range env.Pods {
+		for pvIndex := range put.Spec.Volumes {
+			if put.Spec.Volumes[pvIndex].PersistentVolumeClaim == nil {
 				continue
 			}
-			if statefulset.Spec.ServiceName == localStorage {
-				nonCompliantPods = append(nonCompliantPods, pod)
-				continue
+			for i := range env.PersistentVolumes {
+				if env.PersistentVolumes[i].Spec.StorageClassName == localStorage {
```

Comment:
A couple of issues here:
1. I don't see any relationship here between the `env.PersistentVolumes[i]` and the pod you're checking ("put") on each iteration. Could you explain why are you checking this, please?
2. StorageClasses, as all kubernetes resources, can have any name. Why are you specifically checking against "local-storage"?

Reply:
If local storage exist so I need to check that PVC and also the PV is exist.

In addition, when I did debug I saw the name is local-storage.

@greyerof